File: se/se6029/labs/05-comparison-of-cpp-java/Lab03/Lab03.java
public class Lab03 {
    public static void main(String[] args) {
        Child childObj = new Child(10); // 建立 Child 物件時傳遞參數到父類別建構子
    }
}

class Parent {
    Parent(int x) {
        System.out.println("Parent constructor called with parameter: " + x);
    }
}

class Child extends Parent {
    Child(int y) {
        super(y); // Call parent constructor with parameter
        System.out.println("Child constructor called with parameter: " + y);
    }
}

File: se/se6029/labs/05-comparison-of-cpp-java/Lab03/Lab03.cpp
#include <iostream>

class Parent {
public:
    Parent(int x) {
        std::cout << "Parent constructor called with parameter: " << x << std::endl;
    }
};

class Child : public Parent {
public:
    Child(int y) : Parent(y) {
        std::cout << "Child constructor called with parameter: " << y << std::endl;
    }
};

int main() {
    Child childObj(10); // 建立 Child 物件時傳遞參數到父類別建構子
    return 0;
}
